3p/ampcontext.js

```diff
@@ -109,7 +109,7 @@ export class AmpContext {
    */
   setupMetadata_() {
     try {
-      const data = JSON.parse(decodeURI(this.win_.name));
+      const data = JSON.parse(decodeURI(this.client_.getWindow().name));
```

Comment:
I think we can just save win as a local field in constructor instead of introducing a dependency.